translation/translation.js

```diff
@@ -0,0 +1,271 @@
+                                return $(this).text() === italian
+                            }).html(translationIT_FR[italian]);
+                        }
+                        else {
+                            $("div#content div#switcher.tab-pane div#switcher-layers-container.switcher div.menu-content div#switcher-layers div.without-blocks.no-group table.tree.treeTable tbody tr td span.label").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#switcher.tab-pane div#switcher-baselayer.baselayer div.menu-content div.baselayer-select select option").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div select option").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#map-content div#mini-dock div.tabbable.tabs-below div#mini-dock-content.tab-content div#selectiontool.tab-pane div.selectiontool div.menu-content table tbody tr td select#selectiontool-layer-list option").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#mini-dock div#print.tab-pane div.print div.menu-content table tbody tr td select#print-template option").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#bottom-dock div#attributeLayers.tab-pane div.tab-content div#attribute-layer-list table tbody tr td").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#header div#title h1").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                        }
+                    });
+            });
+
+        }
+    }
+});
+
+//This function is executed when the lizmapeditionformdisplayed event is triggered
+lizMap.events.on({
+    lizmapeditionformdisplayed: function(e) {
+        // get browser language
+        var userLang = navigator.language || navigator.userLanguage;
+        if (userLang === 'fr' || userLang === 'fr-FR'){
+
+            var translationIT_FR2 = {};
+
+            // get the json file with transalted strings
+            // the path to the json file must be provided as relative path, using the web url the browser can return a security error
+            fetch('../../../../../../concerteaux3d/translation.json').then(function(response) {
```

Comment:
You could cache translation.js as you have download it with "uicreated" event above.

Reply:
I have changed the code, now the fetch of the translation.json file is done only once in the "uicreated" event.